src/compiler/scanner.ts

```diff
@@ -1846,6 +1854,10 @@ export function createScanner(languageVersion: ScriptTarget,
                             }
                         }
 
+                        if (isJSDoc && (!skipJSDoc || semanticJSDocTagRegEx.test(text.slice(tokenPos, pos)))) {

```

Comment:
I'm starting with the naive regex first, but given the strings are a fixed size, I will also try the charCodeAt equality check in the loop above too.